File: android/src/main/java/me/haymob/coffeeshop/android/components/LogoutAlert.kt
package me.haymob.coffeeshop.android.components

import androidx.compose.material.AlertDialog
import androidx.compose.material.Button
import androidx.compose.material.Text
import androidx.compose.runtime.Composable

@Composable
fun LogoutAlert(onLogout: () -> Unit, onCancel: () -> Unit) {
    AlertDialog(
        onDismissRequest = onCancel,
        title = { Text("Logout") },
        text = { Text("Do you want to leave?") },
        confirmButton = {
            Button(onClick = onLogout) {
                Text("Logout")
            }
        },
        dismissButton = {
            Button(onClick = onCancel) {
                Text("Cancel")
            }
        }
    )
}

File: ios/ios/Views/LogoutAlert.swift
import SwiftUI

func LogoutAlert(action: @escaping () -> Void) -> Alert {
    Alert(
        title: Text("Logout"),
        message: Text("Do you want to leave?"),
        primaryButton: .cancel(),
        secondaryButton: .default(Text("Logout"), action: action)
    )
}
